python/ray/tune/tests/test_trial_runner_3.py

```diff
@@ -355,6 +355,48 @@ def num_running_trials():
         count = Counter(evaluated)
         assert all(v <= 3 for v in count.values())
 
+    def testTrialErrorResume(self):
+        ray.init(num_cpus=3)
+        tmpdir = tempfile.mkdtemp()
+        runner = TrialRunner(local_checkpoint_dir=tmpdir)
+        kwargs = {
+            "stopping_criterion": {
+                "training_iteration": 4
+            },
+            "resources": Resources(cpu=1, gpu=0),
+        }
+        trials = [
+            Trial("__fake", config={"mock_error": True}, **kwargs),
+            Trial("__fake", **kwargs),
+            Trial("__fake", **kwargs),
+        ]
+        for t in trials:
+            runner.add_trial(t)
+
+        while not runner.is_finished():
+            runner.step()
+
+        runner.checkpoint()
+
+        assert trials[0].status == Trial.ERROR
+        del runner
+
+        new_runner = TrialRunner(
+            rerun_failed=True, resume=True, local_checkpoint_dir=tmpdir)
+        assert len(new_runner.get_trials()) == 3
+        assert Trial.ERROR not in {t.status for t in new_runner.get_trials()}
+        disable_error = False
+        for t in new_runner.get_trials():
+            if t.config.get("mock_error"):
+                t.config["mock_error"] = False
+                disable_error = True
+        assert disable_error
+
+        while not new_runner.is_finished():
+            new_runner.step()
+        assert Trial.ERROR not in {t.status for t in new_runner.get_trials()}
+        shutil.rmtree(tmpdir)
+
```

Comment:
This test also passes when `rerun_failed=False` is set. I think we would want to checkpoint after the trial has failed (as the trials will be PENDING otherwise) and then check how many trials are run with the new runner. Maybe it would make sense to test `rerun_failed=False`, too?